DetectDynamicJS.py

```diff
@@ -70,6 +70,14 @@ def doPassiveScan(self, baseRequestResponse):
         self._requestResponse = baseRequestResponse
         self._helpers = self._callbacks.getHelpers()
         
+        response = self._requestResponse.getResponse()
+        responseInfo = self._helpers.analyzeResponse(response)
+        
+        # Scan only if the statusCode is 200
+        statusCode = responseInfo.getStatusCode()
+        if statusCode != 200:
+          return
```

Comment:
Oh, and also please use `return None` inside this function for style and consistency reasons.